plugins/gatsby-render-components/rendered-component.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+
+let addToRenderQueue = () => undefined
+
+export const setAddToRenderQueueCallback = callback => {
+  addToRenderQueue = callback
+}
+
+const renderedComponent = Component => {
+  return class extends React.Component {
+    constructor(props) {
+      super(props)
+      const { renderOptions, ...passThroughProps } = props
+      const { width, height, relativePath, filename } = renderOptions
+      addToRenderQueue({
+        Component,
+        props: passThroughProps,
+        width,
+        height,
+        relativePath,
+        filename,
+      })
+    }
+
+    render() {
+      const { renderOptions, ...passThroughProps } = this.props
+      return <Component {...passThroughProps} />
```

Comment:
All of these components are going to *just* be used to render PNGs, and not be same as page components. We should just return `null` here.

Reply:
Done